openrtb3/openrtb3.go

```diff
@@ -0,0 +1,162 @@
+}
+
+// Source object carries data about the source of the transaction including the unique ID of the transaction itself, source authentication information, and the chain of custody.
+type Source struct {
+	TID    string          `json:"tid,omitempty"`    // Recommended. Transaction ID that must be common across all participants throughout the entire supply chain of this transaction. This also applies across all participating exchanges in a header bidding or similar publisher-centric broadcast scenario.
+	DS     string          `json:"ds,omitempty"`     // Recommended. Digital signature used to authenticate this request computed by the publisher or its trusted agent from the transaction digest string "tid:digest", where ‘tid’ matches the "tid" attribute and ‘digest’ is a string composed of an immutable portion of domain objects as defined in the domain specification used for this request. Refer to Section “Inventory Authentication” for more details.
+	DSGVer int8            `json:"dsgver,omitempty"` // Recommended. Format version of the digest string that was digitally signed to produce “ds”. Refer to Section “Inventory Authentication” for more details.
+	Digest string          `json:"digest,omitempty"` // The full transaction digest string that was signed to produce the digital signature. Refer to Section “Inventory Authentication” for more details. NOTE: This is only intended for debugging purposes as needed. It is not intended for normal Production traffic due to the bandwidth impact.
+	Cert   string          `json:"cert,omitempty"`   // Recommended. File name of the certificate (i.e., the public key) used to generate the digital signature in “ds” attribute. Refer to Section “Inventory Authentication” for more details.
+	PChain string          `json:"pchain,omitempty"` // Payment ID chain string containing embedded syntax described in the TAG Payment ID Protocol. NOTE that the authentication features in this Source object combined with the “ads.txt” specification may lead to the future deprecation of this attribute.
+	Ext    json.RawMessage `json:"ext,omitempty"`    //	Optional exchange-specific extensions.
+}
+
+// Item object represents a unit of goods being offered for sale either on the open market or in relation to a private marketplace deal.
+// The id attribute is required since there may be multiple items being offered in the same bid request and bids must reference the specific item of interest.
+// This object interfaces to Layer-4 domain objects for deeper specification of the item being offered (e.g., an impression).
+type Item struct {
+	ID      string          `json:"id"`                // Required. A unique identifier for this item within the context of the offer (typically starts with 1 and increments).
+	Qty     int             `json:"qty,omitempty"`     // Default: 1. The number of instances (i.e., "quantity") of this item being offered (e.g., multiple identical impressions in a digital out-of-home scenario).
```

Comment:
Document when `int` may be used (when it can be used as index or compared to `len(smth)`)